### ISSUE TITLE
Scope options for accessibility / make initial fixes

### ISSUE THREAD

**Issue by rachelhbd** [350org/ak_intl_v3#361]
Hey, @cjmabry please could you share your accessibility recommendations for donate pages with us? Do you think they are quick fixes we could do in August, or if more significant could be something we'd scope in Aug and work on in Sept?

Also if you have a link to the accessibility checker you used, that would be v helpful for future!

**Comment by cjmabry** [comment]
Here is a link to an accessibility report for the main web donation page - I'll summarize my takeaways below: https://wave.webaim.org/report#/https://act.350.org/donate/web_donate_button/

While these automated tools are good to check, some amount of manual testing should also be done, including:
- testing if site is completely navigable only with keyboard manually by tabbing through items
- viewing site without styles
- testing with screenreader software and other accessibility tools. Browserstack and other platforms can help (https://www.browserstack.com/accessibility-testing)

Happy to do a full audit if you think necessary.

You can test pages in future via web browser here (https://wave.webaim.org/), or install Chrome extension.

**Report Summary**

**Errors**
These are the most major items flagged and should be looked at first.

- [ ] **Missing alt text on an image:** the Charity Navigator logo in the footer does not have alt text, so a screenreader wouldn't be able to explain to the user what is there. This should be a simple, one-line fix to an an alt tag on the image, but since it's a minor image in the footer, it's not a huge deal.
- [ ] **Missing form labels:**  Some form labels on the Action Kit form don't have labels, so a screenreader might not be able to describe what a form was. This is could be a bigger issue, as users would basically be in the dark on what they're filling in, or it could be a non-issue if the affected field is not meant for user readability. Further tests with actual screenreader would probably be necessary to determine if this is a big issue or not. I would recommend testing with a real screen reader or Browserstack to determine if it is. I can do this if you'd like.

**Contrast Errors**
It's a good idea to minimize contrast errors as it makes text harder to read for all.

- [ ] **Contrast issues with footer text:** A simple solution to fix would be to either make the background darker in the footer, or to make the text bigger in the footer, or a combination of both, since contrast is related to size of text. You can see what values would work by inputting the background and foreground colors here: https://webaim.org/resources/contrastchecker/
- [ ] **Image background contrast:** It's worth noting that the contrast checker doesn't check text over gradient or image backgrounds, but the contrast of the background image and text on the main part of the form might not be sufficient. It's best to have solid colors behind text that pass checks. Solving this would probably require some design thinking. See more here: https://www.w3.org/WAI/WCAG21/Techniques/failures/F83

**Alerts**
Alerts are good to look at but should be taken in context and might not necessarily be big issues. 

- [ ] **Missing labels and other form alerts:** Most of these issues are related to HTML syntax issues - basically, not having an H1 and skipping to a lower level heading, or not having form labels referencing the appropriate forms, or having an item behave as a heading but be a <p> tag. I can break these down into individual issues if you'd like, but I would say probably worth starting with above errors first. We should also dig further about what control we have over messing with the Action Kit form fields.

@rachelhbd let me know if you have any questions or would like me to expand on anything!

**Comment by rachelhbd** [comment]
Thanks very much @cjmabry! So for the "Image background contrast" is the contrast checker essentially not seeing/taking into account the combined effect of the background image + the gradient there?

Although I guess that if even the plain white text on blue background is failing, then the gradient portion would already be similar or worse

@itsboit @stbensonimoh this is something we could discuss!

**Comment by cjmabry** [comment]
@rachelhbd Yes, that is my understanding - the tool doesn't check text over gradients and images, presumably because there are many different points of contrast. You could look at a black and white version of the page to get a feel for contrast, but you're right, if the plain text with blue is failing, this probably is, too, although the text is much bigger which might offset that.

**Comment by rachelhbd** [comment]
yep, that makes sense thanks - will chat with Cliff on this initially

**Comment by rachelhbd** [comment]
@cjmabry looping back here! 

I think let's prioritise visual accessibility for Sept initially (tackle the contrast errors) as that would be something we likely need to A/B test, whereas adding labels etc. will be something we can directly implement without needing a testing window. 

On reflection I'm curious about what you might recommend for the design if we did want to add more of a white background, would you be up for thinking about this and maybe putting a design/mockup together for us to get some feedback on?

I was looking for the test I thought we ran with a plain white background vs current blue fade, but can't find it so may have imagined it! But we know at least that [this page](https://secure.actblue.com/donate/jun-22-em-us-climate-disasters-redirect/ ) outperforms our AK donate pages, probably for multiple reasons -- so you might be right that this is a better approach. For branding consistency, we may want to keep a fade over the image still at least for now (or test that too)

I'm thinking more about the concept of embedding the form in the Sign > donate > share flow as well - so was musing that it might be good to be thinking about both tracks at the same time

**Comment by rachelhbd** [comment]
On the keyboard navigable point - I know that our donate forms are *not* fully keyboard navigable due to this bugfix: https://gitlab.com/350/product-fa-sprints/-/issues/17 
but I think AK has fixed the original cause of the problem so we could now revert this - might be a quickish fix for Sept as well

**Comment by cjmabry** [comment]
@rachelhbd I can come up with some designs here by next week!

Off the top of my head, I would propose two potential solutions:

1. Keep the gradient and image background but make the gradient overlay white, and change colors of copy and forms to pass contrast
2. Go full white background over text, reverse colors, and do photo elsewhere (similar to Actblue link you shared)

I have seen past client test full background vs. white actblue background and white won, but isolated case, and could also be due to pagespeed (i.e. not loading an image)

**Comment by cjmabry** [comment]
> On the keyboard navigable point - I know that our donate forms are _not_ fully keyboard navigable due to this bugfix: https://gitlab.com/350/product-fa-sprints/-/issues/17 but I think AK has fixed the original cause of the problem so we could now revert this - might be a quickish fix for Sept as well

@rachelhbd looks like I don't have access here under chris@withtheranks.com

**Comment by rachelhbd** [comment]
Great, thanks @cjmabry! We actually have a white version of the background available which we could try testing for 1. - looks like there are only a few bits of blue on white text that might need to be larger https://wave.webaim.org/report#/https://act.350.org/donate/month-year-em-eu-description_copy8/ 
How about I talk with Maria about scheduling an A/B test for the existing version to see how that performs, and if it passes we can tweak the small bits of text that are failing? Or would you suggest any changes to it before we test?

For 2, I'm definitely curious to see what you're thinking this might look like - we could take some time to explore design options during Sept aiming for the Oct sprint for the next batch of dev?

Also you should have an invite to gitlab, thanks for the flag!

**Comment by rachelhbd** [comment]
Just as a thought, here's a recent-ish design from Cliff for the unsubscribe page https://act.350.org/act/unsubscribe-new 
Could be a useful reference for a non-full-bleed white background with some kernels - asking if he has any more reference points for us too!

**Comment by rachelhbd** [comment]
Closing this as have [designs in](https://www.figma.com/proto/9o0tUU1krckXCBLzxjozB0/350-Donation-Accessibility-Ideation?node-id=502%3A453&viewport=3311%2C-1900%2C0.28&scaling=scale-down&page-id=0%3A1%3Ffuid%3D923664738342119661) for the visual accessibility, with all text passing with some question marks/difficulty gauging with the darkened gradient one (#2)
